### PR TITLE
Allow <Banner> usage in app

### PR DESCRIPTION
**Pull request by fregante**
- For https://github.com/pixiebrix/pixiebrix-app/pull/257

### CODE REVIEW

File: src/messaging/external.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ import { isChrome } from "@/helpers";
 const lift = isChrome ? liftBackground : liftExternal;
 
 export const connectPage = lift("CONNECT_PAGE", async () => {
-  return browser.runtime.getManifest();
+  return chrome.runtime.getManifest();
```
Comment (fregante) [inline src/messaging/external.ts:32]: `browser`’s types don't currently return the whole manifest. I've been using `chrome` to fix this issue

File: src/layout/Banner.tsx [hunk 2]
```diff
@@ -45,16 +47,28 @@ function useSyncedHostname() {
   return hostname;
 }
 
+function useVersionName() {
+  const [versionName, setVersionName] = useState<string>();
+
+  useAsyncEffect(async () => {
+    const manifest = isExtensionContext()
```
Comment (fregante) [inline src/layout/Banner.tsx:54]: I suppose this would be a dependency but it's constant

File: src/layout/Banner.tsx [hunk 3]
```diff
@@ -45,16 +47,28 @@ function useSyncedHostname() {
...
+    setVersionName(manifest.version_name);
+  }, []);
+
+  return versionName;
+}
+
 const Banner: React.FunctionComponent = () => {
   const { extension } = useContext(AuthContext);
   const hostname = useSyncedHostname();
+  const versionName = useVersionName();
```
Comment (fregante) [inline src/layout/Banner.tsx:66]: I find it weird that these 2 hooks are called even on production, but if I move them lower I get

```
ESLint: React Hook "useSyncedHostname" is called conditionally. React Hooks must be called in the exact same order in every component render. (react-hooks/rules-of-hooks)
```

Comment (twschiller) [reply src/layout/Banner.tsx:66]: No, that's actually a real error. React Hooks use a bit of magic (I can't explain it very well), that actually require them all to be called in the exact same order every time. Although some of them (e.g., useEffect) has support for dependency lists to determine if they're a noop when run

Comment (fregante) [reply src/layout/Banner.tsx:66]: > magic

😰

Comment (fregante) [reply src/layout/Banner.tsx:66]: What I mean is that due to this, more code is executed than necessary in this case, I think. That seems like a real problem that shouldn't be enforced by a framework, so I hope the magic is also avoiding that.

Comment (twschiller) [reply src/layout/Banner.tsx:66]: Haha, any advanced technology is indistinguishable from magic: https://www.netlify.com/blog/2019/03/11/deep-dive-how-do-react-hooks-really-work/

More that I just haven't looks into how they're implemented under the hood

**Review by twschiller — APPROVED** [review]
I refactored the code to use our `useAsyncState` helper hook, which handles the pattern of useAsyncEffect + useState

I also split out the DevelopmentBanner from the main Banner which is just a noop in production mode

**Pull request merged**